pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -667,6 +683,36 @@ File untarAndMoveSegment(String segmentName, File tarFile, File tempRootDir)
     }
   }
 
+  @VisibleForTesting
+  File untarAndMoveSegmentWithRetries(String segmentName, File tarFile, File tempRootDir, SegmentZKMetadata zkMetadata)
+      throws AttemptsExceededException {
+    AtomicReference<File> file = new AtomicReference<>();
+    try {
+      file.set(untarAndMoveSegment(segmentName, tarFile, tempRootDir));
+    } catch (Exception e) {
+      try {
+        RetryPolicies.exponentialBackoffRetryPolicy(_retryCount, _retryWaitMs, _retryDelayScaleFactor).attempt(() -> {
+          try {
+            File retriedTarFile = downloadAndDecrypt(segmentName, zkMetadata, tempRootDir);
```

Comment:
(not sure if this is needed) Maybe we need to clean up previously downloaded files in order to not use up disk space?

Reply:
Thanks, that is a valid observation.
I am now deleting the retriedTarFile in finally block.